search/includes/classes/class-versioning.php

```diff
@@ -221,15 +226,14 @@ public function get_versions( Indexable $indexable, bool $provide_default = true
 		$versions = [];
 
 		if ( $indexable->global ) {
-			$versions = get_site_option( self::INDEX_VERSIONS_OPTION_GLOBAL, array() );
+			$versions = get_site_option( self::INDEX_VERSIONS_OPTION_GLOBAL );
```

Comment:
What's the reason behind changing the default value to empty string here?

Reply:
Isn't it being changed to `false`? And in regards for the _why_ it is being changed to `false` is because we have a way of knowing that the index versioning hasn't been touched yet, so the option will not exist at that point and then, we can use that later on to determine whether to return `1` as a default version versus just not having any indexes (albeit, if they get in a state where both are deleted).

Reply:
You're right, it's `false`, I thought it's `''` (an empty string). My question was more around the fact we're expecting `$versions` to be an array.

According to my testing, on a fresh dev-env instance this option is present:

```
wp> get_option( 'vip_search_index_versions' );
=> array(1) {
  ["post"]=>
  array(1) {
    [1]=>
    array(4) {
      ["number"]=>
      int(1)
      ["active"]=>
      bool(true)
      ["created_time"]=>
      NULL
      ["activated_time"]=>
      NULL
    }
  }
}
```

Reply:
furhtermore, IF I delete it it get's recreated on the next request:

```
wp> delete_option( 'vip_search_index_versions' );
=> bool(true)
wp> get_option( 'vip_search_index_versions' );
=> bool(false)
wp> exit
 ~  vip dev-env exec --slug search -- wp shell
Running validation steps...
✓ Check for docker installation
✓ Check for docker-compose installation
✓ Check access to docker for current user
✓ Check DNS resolution
wp> get_option( 'vip_search_index_versions' );
=> array(1) {
  ["post"]=>
  array(1) {
    [1]=>
    array(4) {
      ["number"]=>
      int(1)
      ["active"]=>
      bool(true)
      ["created_time"]=>
      NULL
      ["activated_time"]=>
      NULL
    }
  }
}
```

Reply:
Hmmm on a fresh dev-env instance where the versioning hasn't been touched at all, I'm getting:
```
wp> get_option( 'vip_search_index_versions' );
=> bool(false)
```

Reply:
Same happens not-locally, and for multiple versions it even detects it correctly.:

```
wp> get_option( 'vip_search_index_versions' );
=> phar:///usr/local/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php:52:
array(1) {
  'post' =>
  array(2) {
    [1] =>
    array(4) {
      'number' =>
      int(1)
      'active' =>
      bool(true)
      'created_time' =>
      NULL
      'activated_time' =>
      NULL
    }
    [2] =>
    array(4) {
      'number' =>
      int(2)
      'active' =>
      bool(false)
      'created_time' =>
      int(1644466599)
      'activated_time' =>
      NULL
    }
  }
}
wp> update_option( 'vip_search_index_versions_backup', get_option( 'vip_search_index_versions' ) );
=> phar:///usr/local/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php:52:
bool(true)
wp> get_option( 'vip_search_index_versions' );
=> phar:///usr/local/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php:52:
bool(false)
root@cantina-trunk-staging-sbx-u139-54bb787cd9-hppgz:~# wp shell

wp> get_option( 'vip_search_index_versions' );
=> phar:///usr/local/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php:52:
bool(false)
root@cantina-trunk-staging-sbx-u139-54bb787cd9-hppgz:~# wp shell
wp> get_option( 'vip_search_index_versions' );
=> phar:///usr/local/bin/wp/vendor/wp-cli/shell-command/src/WP_CLI/Shell/REPL.php:52:
array(1) {
  'post' =>
  array(2) {
    [1] =>
    array(4) {
      'number' =>
      int(1)
      'active' =>
      bool(true)
      'created_time' =>
      NULL
      'activated_time' =>
      NULL
    }
    [2] =>
    array(4) {
      'number' =>
      int(2)
      'active' =>
      bool(false)
      'created_time' =>
      NULL
      'activated_time' =>
      NULL
    }
  }
}
```